### PR TITLE
Remove fontawesome

### PR DESCRIPTION
**Pull request by JKarlavige**
## What are you changing in this pull request and why?
[Asana task](https://app.asana.com/0/1200099998847559/1205952377161421/f)

This removes Font Awesome as part of the effort to reduce our Font Awesome bandwidth usage. We didn't use Font Awesome much, and these sections have been updated to use SVGs instead.

## Previews

Social icons in footer:
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/

Social icons for authors:
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/author/joel_labes
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/author/doug_guthrie
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/author/emily_riederer

Guide card icons (search icon, external link icons, fa icons on cards in bullets below):
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/guides
- Quickstart for dbt Core using GitHub Codespace
- Quickstart for dbt Core from a manual install

Individual guide prev/next arrows and mobile menu caret:
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/guides/adapter-creation?step=5

Available integrations (external link icon on cards:
https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/docs/use-dbt-semantic-layer/avail-sl-integrations

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -262,8 +267,7 @@ var siteSettings = {
     "https://cdn.jsdelivr.net/npm/clipboard@2.0.11/dist/clipboard.min.js",
     "/js/headerLinkCopy.js",
     "/js/gtm.js",
-    "/js/onetrust.js",
-    "https://kit.fontawesome.com/7110474d41.js",
```
Comment (JKarlavige) [inline website/docusaurus.config.js:266]: Removing font awesome script

File: website/snippets/_sl-partner-links.md
Comment (JKarlavige) [inline website/snippets/_sl-partner-links.md:1]: This update affects the external link icon on the cards on this page: https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/docs/use-dbt-semantic-layer/avail-sl-integrations

File: website/src/components/author/index.js [hunk 2]
```diff
@@ -28,49 +29,63 @@ function Author(props) {
...
+                        title={`${name} - Social`}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                        key={i}
+                      >
+                        {/* <!--! Font Awesome Pro 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --> */}
+                        {link?.icon ? (
+                          <div className="social-icon">
+                            {getSvgIcon(link?.icon)}
+                          </div>
```
Comment (JKarlavige) [inline website/src/components/author/index.js:72]: Now uses svgs for author social links: https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/author/emily_riederer

File: website/src/components/quickstartGuideCard/index.js
Comment (JKarlavige) [inline website/src/components/quickstartGuideCard/index.js:1]: Affects right arrow next to `Start` on Guide cards: https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/guides

File: website/src/components/quickstartTOC/index.js
Comment (JKarlavige) [inline website/src/components/quickstartTOC/index.js:1]: Affects prev/next buttons on guide page: https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/guides/codespace?step=1

File: website/src/components/searchInput/index.js
Comment (JKarlavige) [inline website/src/components/searchInput/index.js:1]: Affects search icon on guides page: https://docs-getdbt-com-git-remove-fontawesome-dbt-labs.vercel.app/guides

**Review by breezyfasano — APPROVED** [review]
LGTM :shipit:

**Review by john-rock — APPROVED** [review]
:shipit:

**Pull request merged**